sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala

```diff
@@ -376,14 +376,10 @@ class HDFSFileCatalog(
         HadoopFsRelation.shouldFilterOut(name)
       }
 
-      val (dirs, files) = statuses.partition(_.isDirectory)
+      val (_, files) = statuses.partition(_.isDirectory)
 
       // It uses [[LinkedHashSet]] since the order of files can affect the results. (SPARK-11500)
-      if (dirs.isEmpty) {
-        mutable.LinkedHashSet(files: _*)
-      } else {
-        mutable.LinkedHashSet(files: _*) ++ listLeafFiles(dirs.map(_.getPath))
-      }
+      mutable.LinkedHashSet(files: _*)
```

Comment:
Are you sure of the difference between 1.6.1 and master? I see this logics are not changed comparing to that [interfaces.scala#L467-L472](https://github.com/apache/spark/blob/branch-1.6/sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala#L467-L472)
Also, does this still support to read [partitioned tables](http://spark.apache.org/docs/latest/sql-programming-guide.html#partition-discovery)?